src/main/java/org/spongepowered/api/world/explosion/ResistanceCalculator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world.explosion;
+
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.block.BlockState;
+
+/**
+ * A Functional Interface for the fetching of a Block's explosion resistance during an explosion
+ */
+@FunctionalInterface
+public interface ResistanceCalculator {
+    /**
+     * A blueprint for a lambda that takes three variables; BlockState, Vector3i, and Explosion and returns a float.
+     *
+     * @param blockState    The state of the block
+     * @param blockPosition The position of the block in the World
+     * @param explosion     The explosion affecting the block
+     * @return The explosion resistance of the block
+     */
+    float calculateResistance(final BlockState blockState, final Vector3i blockPosition, final Explosion explosion);
```

Comment:
It's not obvious how to get a `World` here. Should the block position be a `Location<World>` instead, and drop the block state (which you can get from the location - fairly standard thing we do?)

What about if a plugin wants to affect obsidian blocks only? How would they determine the resistance for other blocks?

Earlier, I said:

> a `ResistanceCalculator` which takes a Location, **the current resistance as a float**, and returns the new resistance as a float.

That's the reason why I suggested it.

Reply:
> It's not obvious how to get a `World` here. Should the block position be a `Location<World>` instead, and drop the block state (which you can get from the location - fairly standard thing we do?)

If that's the standard way of doing it then I'll rewrite that section.

> What about if a plugin wants to affect obsidian blocks only? How would they determine the resistance for other blocks?
> 
> Earlier, I said:
> 
> > a `ResistanceCalculator` which takes a Location, **the current resistance as a float**, and returns the new resistance as a float.
> 
> That's the reason why I suggested it.

Uh... Mistakes have been made 🤕 . I'll review this and get back to you.

Reply:
I mean, you can keep the block state in, to be honest, it's already been determined. It's more important that the default resistance is provided.

Reply:
I believed it would be more optimized to pass it than fetch it again.
The world can be obtained by either `event.getTargetWorld()` or `explosion.getWorld`.